async-nats/src/jetstream/kv/mod.rs

```diff
@@ -1026,21 +1026,32 @@ impl Store {
     /// # }
     /// ```
     pub async fn keys(&self) -> Result<Keys, HistoryError> {
-        let subject = format!("{}>", self.prefix.as_str());
+        self.keys_with_filters(vec![">"]).await
+    }
 
-        let consumer = self
-            .stream
-            .create_consumer(super::consumer::push::OrderedConfig {
-                deliver_subject: self.stream.context.client.new_inbox(),
-                description: Some("kv history consumer".to_string()),
-                filter_subject: subject,
-                headers_only: true,
-                replay_policy: super::consumer::ReplayPolicy::Instant,
-                // We only need to know the latest state for each key, not the whole history
-                deliver_policy: DeliverPolicy::LastPerSubject,
-                ..Default::default()
-            })
-            .await?;
+    pub async fn keys_with_filters(&self, filters: Vec<&str>) -> Result<Keys, HistoryError> {
```

Comment:
I think I would go with `impl IntoIterator` here for better flexibility.

Reply:
I think I did here, but again as I'm a newbie on rust I'd love you to check if I did things right. I didn't want to waste loops to count stuff, so I basically looped through the iterator just once. If I detect we're at index 1 (meaning len() is > 1) then I move the `filter_subject` to `filter_subjects`
Also: I used match {} to `switch` between cases for the current index

Reply:
Close ;).

It should be `impl IntoIterator` instead of `Vec<&str>`

Reply:
it should already be, I think the code up here is outdated, this is the new one:

https://github.com/nats-io/nats.rs/blob/6a9c6151cb768e727c738c87e607af7cfad06fba/async-nats/src/jetstream/kv/mod.rs#L1032-L1035